src/components/Pagination/Pagination.jsx

```diff
@@ -0,0 +1,208 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import { Flex, PseudoBox, useTheme } from '@chakra-ui/core';
+
+const LEFT_PAGE = 'LEFT';
+const RIGHT_PAGE = 'RIGHT';
+
+/**
+ * @function Pagination
+ *
+ * @param {number} totalRecords - Total records count
+ * @param {function} onPageChanged - Callback function that passes paginated information
+ * @param {number} pageLimit - Number of items to display per page
+ * @param {number} pageNeighbors - Number of neighboring pages when pagination is in the middle of the list
+ *
+ */
+
+const Pagination = props => {
+  const theme = useTheme();
+
+  const [totalRecords, setTotalRecords] = useState(null);
+  const [totalPages, setTotalPages] = useState(0);
+  const [pageLimit, setPageLimit] = useState(10);
+  const [pageNeighbors, setPageNeighbors] = useState(1);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [pages, setPages] = useState([]);
+
+  useEffect(() => {
+    const { totalRecords, pageLimit, pageNeighbors } = props;
+
+    if (pageLimit) {
+      setPageLimit(pageLimit);
+    }
+    if (pageNeighbors) {
+      setPageNeighbors(Math.max(0, Math.min(pageNeighbors, 2)));
+    }
+
+    const totalPages = Math.ceil(totalRecords / pageLimit);
+    setTotalRecords(totalRecords);
+    setTotalPages(totalPages);
+  }, []);
```

Comment:
Uhh, there's a warning here about not providing the `totalRecords`, `pageLimit`, and `pageNeighbors` values from `props` to the dependencies array. That should probably have been addressed before merging as that opens up the code to possible bugs in edge cases.

Reply:
Thanks for your comments @ChrisBrownie55! I can work on a fix to address these errors but I believe someone is currently working on tweaking the pagination buttons.

I can wait until after they are done to see if they have already fixed it.